### PR TITLE
replaces bloodcrawl spell with ctrl+click

### PR DESCRIPTION
**Pull request by ShadowDeath6**
:cl: ShadowDeath6
rscadd: Replaced the bloodcrawl spell for slaughter demons with the ability to ctrl+click blood.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 1]
```diff
@@ -47,12 +47,12 @@
 	del_on_death = 1
 	deathmessage = "screams in anger as it collapses into a puddle of viscera!"
 
-/mob/living/simple_animal/slaughter/Initialize()
+/*/mob/living/simple_animal/slaughter/Initialize()
```
Comment (ThatLing) [inline code/modules/antagonists/slaughter/slaughter.dm:50]: Don't comment this, just override it in a different file

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 2]
```diff
@@ -1,8 +1,8 @@
 //////////////////The Monster
 
 /mob/living/simple_animal/slaughter
-	name = "slaughter demon"
-	real_name = "slaughter demon"
+	name = "Slaughter Demon"
```
Comment (ThatLing) [inline code/modules/antagonists/slaughter/slaughter.dm:4]: Why?

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 3]
```diff
@@ -71,7 +71,11 @@
 /mob/living/simple_animal/slaughter/phasein()
 	. = ..()
 	speed = 0
-	boost = world.time + 60
+	boost = world.time + 20
+
+/mob/living/simple_animal/slaughter/bloodpool_sink(obj/effect/decal/cleanable/B)
```
Comment (ThatLing) [inline code/modules/antagonists/slaughter/slaughter.dm:76]: Put this in a /yogstation file

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 4]
```diff
@@ -71,7 +71,11 @@
 /mob/living/simple_animal/slaughter/phasein()
 	. = ..()
 	speed = 0
-	boost = world.time + 60
+	boost = world.time + 20
```
Comment (ThatLing) [inline code/modules/antagonists/slaughter/slaughter.dm:74]: Mark your changes like this

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 5]
```diff
@@ -90,25 +94,23 @@
 	user.visible_message("<span class='warning'>[user] raises [src] to their mouth and tears into it with their teeth!</span>", \
 						 "<span class='danger'>An unnatural hunger consumes you. You raise [src] your mouth and devour it!</span>")
 	playsound(user, 'sound/magic/demon_consume.ogg', 50, 1)
-	for(var/obj/effect/proc_holder/spell/knownspell in user.mind.spell_list)
-		if(knownspell.type == /obj/effect/proc_holder/spell/bloodcrawl)
-			to_chat(user, "<span class='warning'>...and you don't feel any different.</span>")
-			qdel(src)
-			return
+	if(user.bloodcrawl == BLOODCRAWL)
+		to_chat(user, "<span class='warning'>...and you don't feel any different.</span>")
```
Comment (ThatLing) [inline code/modules/antagonists/slaughter/slaughter.dm:98]: Mark your changes like this

File: code/modules/mob/living/bloodcrawl.dm [hunk 6]
```diff
@@ -22,25 +22,27 @@
...
-		B1.icon_state = "bloodhand_left"
-		B2.icon_state = "bloodhand_right"
-		C.put_in_hands(B1)
-		C.put_in_hands(B2)
-		C.regenerate_icons()
-	src.notransform = TRUE
-	spawn(0)
-		bloodpool_sink(B)
-		src.notransform = FALSE
+	var/turf/bloodloc = get_turf(B.loc)
```
Comment (ThatLing) [inline code/modules/mob/living/bloodcrawl.dm:25]: Just override this proc in a /yogstation file at this point

File: code/modules/mob/living/living.dm [hunk 7]
```diff
@@ -286,6 +286,8 @@
 
 	if(istype(AM) && Adjacent(AM))
 		start_pulling(AM)
+	else if(bloodcrawl == BLOODCRAWL_EAT) //so demons can ctrl + click tiles to teleport to them
```
Comment (ThatLing) [inline code/modules/mob/living/living.dm:289]: Mark your changes like this

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 8]
```diff
@@ -48,6 +48,7 @@ Difficulty: Hard
 	var/charging = FALSE
 	medal_type = BOSS_MEDAL_BUBBLEGUM
 	score_type = BUBBLEGUM_SCORE
+	bloodcrawl = BLOODCRAWL_EAT
```
Comment (ThatLing) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:51]: Mark your changes like this

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 9]
```diff
@@ -107,10 +108,10 @@ Difficulty: Hard
 	for(var/mob/living/simple_animal/hostile/megafauna/bubblegum/B in GLOB.mob_living_list)
 		if(B != src)
 			return INITIALIZE_HINT_QDEL //There can be only one
-	var/obj/effect/proc_holder/spell/bloodcrawl/bloodspell = new
+	/*var/obj/effect/proc_holder/spell/bloodcrawl/bloodspell = new
```
Comment (ThatLing) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:111]: Mark your changes like this

File: yogstation/code/modules/spells/spell_types/bloodcrawl.dm [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+	revert_cast()
+	to_chat(user, "<span class='warning'>There must be a nearby source of blood!</span>")
+
+/obj/effect/proc_holder/spell/bloodcrawl/perform(obj/effect/decal/cleanable/target, recharge = 1, mob/living/user = usr)
+	if(istype(user))
+		if(phased == 1)
+			if(user.phasein(target))
+				phased = 0
+		else
+			if(user.phaseout(target))
```
Comment (ThatLing) [inline yogstation/code/modules/spells/spell_types/bloodcrawl.dm:30]: Remove the commented code

**Review by ThatLing — CHANGES_REQUESTED** [review]

**Comment by ShadowDeath6** [comment]
closing this until I'm not busy

**Pull request closed without merging**